### ISSUE TITLE
[Bug]: Checkbox appears red when widget name starts with digits

### ISSUE THREAD

**Issue by shwetha-ramesh** [appsmithorg/appsmith#22259]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

1. Checkbox appears red when widget name starts with digits (say 1checkbox) in checkbox group widget
2. Selected value disappears and placeholder is displaced when widget name starts with digits (say 1Select) in Select family widgets

![image](https://user-images.githubusercontent.com/92293815/231085919-c17311c0-9379-4a6b-af74-cd297a1e96f1.png)

### Steps To Reproduce

1. Drop a checkbox group widget > Rename the widget to 1checkbox
2. Drop a Select/multiselect/multitree/tree widget > Rename the widget to 1Select

### Environment

Production

### Issue video log

https://www.loom.com/share/d652669b929e455086e8ee07401b980f

### Version

Cloud

**Comment by iSatVeerSingh** [comment]
is this issue still unresolved?

**Comment by saicharan-zemoso** [comment]
Hello @Nikhil-Nandagopal can I try to resolve this issue?

**Comment by saicharan-zemoso** [comment]
I want clarity in this.
1. the checkbox turning red is a issue or The checkBox group widget is not showing the error.
2. When the checkbox group is renamed and if that starts with number I could not use that entity
this is what happening if the widget name start with number:
![image](https://github.com/user-attachments/assets/dac4c369-9e5b-4d0a-9266-a36be50fbf88)


Is this an expected behaviour??
If this is an expected behaviour then the real bug will be that the checkboxs are turning reb but not showing any warning and error.
pic of no error or warning when the widget name starts with number.

**Comment by akshayvijayjain** [comment]
so original bug can be==> while renaming/ naming a widget to something that starts with number, it should throw error.

**Comment by Nikhil-Nandagopal** [comment]
@akshayvijayjain @saicharan-zemoso Yes the correct fix for this should be to prevent a widget or query name from starting with a digit.
